packages/components/src/components/Icon/Icon.tsx

```diff
@@ -1,21 +1,29 @@
-import { ColorValue, useWindowDimensions } from 'react-native'
+import { ColorValue, View, ViewStyle, useWindowDimensions } from 'react-native'
 import { SvgProps } from 'react-native-svg'
 import React, { FC } from 'react'
 
 import { IconMap } from './iconList'
 import { useColorScheme, useTheme } from '../../utils'
 
-type nameOrSvg =
+type nameOrSvgOrNoIcon =
   | {
       /** Name of preset icon to use {@link IconMap} **/
       name: keyof typeof IconMap
+      noIcon?: never
```

Comment:
The addition of the `noIcon` prop works, but what are your thoughts on allowing a `'none'` value to be passed to the `name` prop to achieve the same thing without the addition of a new prop.

Reply:
I originally tried it that way with having name be `[existing typing] | 'no icon'`, but it was causing a huge headache because although the name list seems like strings when calling, it's actually the SVG-mapping components (`React.FC<SvgProps>`) which is an object.

Most notably this was causing issues around trying to pass the IconProps sent by the consumer because the spread, `{..._icon}`, to pass the property overrides (if present) didn't want to allow conditionalizing when the name prop was an object or string. Trying to manually set the name and then `..._icon` the rest then also ran into issues with the exclusionary typing for name and svg props. 

Ultimately, the headache was proving sufficiently annoying to try resolving that I changed it up. One key factor was: I didn't want to pull it out of what is passed to the Icon to do more meaningful manipulation of the props as that seemed like it would weaken the defensiveness by have more non-render-time conditionality that this defensive change was specifically trying to avoid.

Reply:
Okay let's keep the prop then. Doesn't seem worth the headache. It'll probably be used sparingly, maybe internally only.

Reply:
Thinking more on this, it would potentially work fine if actually adding `'none'` directly to IconMap--in retrospect, perhaps the object it was referring to wasn't the `React.FC<SvgProps>` but the object of all the icons typed that way. But that would involve a weird edit to the buildIcons script and not sure what it'd be typed as or if that'd cause other problems with it not being `React.FC<SvgProps>` like all the other icons in IconMap.